bolt_unix.go

```diff
@@ -12,6 +12,59 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+// pwritev writes the given pages using vectorized io. The pages must be sorted by their id indicating their sequence.
+// The logic identifies runs of consecutive pages that are written with vectorized io using the pwritev syscall.
+func pwritev(db *DB, pages []*page) error {
+	offsets, iovecs := pagesToIovec(db, pages)
+
+	var (
+		e syscall.Errno
+	)
+	for i := 0; i < len(offsets); i++ {
+		_, _, e = syscall.Syscall6(syscall.SYS_PWRITEV, db.file.Fd(), uintptr(unsafe.Pointer(&iovecs[i][0])),
+			uintptr(len(iovecs[i])), uintptr(offsets[i]), uintptr(offsets[i]>>0x8), 0)
+		if e != 0 {
+			return e
+		}
+	}
+
+	return nil
+}
+
+func pagesToIovec(db *DB, pages []*page) ([]uint64, [][]syscall.Iovec) {
+	var offsets []uint64
+	var iovecs [][]syscall.Iovec
+
+	// TODO: read from this from sysconf(_SC_IOV_MAX)?
+	// linux and darwin default is 1024
+	const maxVec = 1024
+
+	lastPid := pages[0].id - 1
+	begin := 0
+	var curVecs []syscall.Iovec
+	for i := 0; i < len(pages); i++ {
+		p := pages[i]
+		if p.id != (lastPid+1) || len(curVecs) >= maxVec {
```

Comment:
This seems not correct. Shouldn't it be something like below? 
```
p.id != lastPid + last_overflow + 1
```

Reply:
interesting, let me see how the overflow works. I was under the impression that the page id is just an increasing identifier for the page.

Reply:
fixed, very curious that no e2e test case covered this. I'm also not even sure whether the overflow is in use.